daemon_unix.go

```diff
@@ -115,6 +116,71 @@ func (d *Context) parent() (child *os.Process, err error) {
 	d.rpipe.Close()
 	encoder := json.NewEncoder(d.wpipe)
 	err = encoder.Encode(d)
+
+	// wait for worker to start running or else network calls fail: https://github.com/sevlyar/go-daemon/issues/100
+	time.Sleep(2 * time.Second)
```

Comment:
Thank you for the contribution.

The solution is not stable and makes the bug is noise. Any delay leaves chance for reproducing of the bug. 

We should implement child-parent processes syncronization to fix the bug properly. Child process can write an acknowledgment to its parent after initialization and parent can wait the acknowledgment.

Reply:
Hey @sevlyar, sorry about the extra changes. Forgot to remove those. 

I tried child parent synchronization and it still doesn't work 😞

It seems like it has to have some sort of activity in worker happen first before parent exits or it doesn't let anything else happen.

But! I likely didn't do it right. Are you able to help me with the synchronization? I am unfortunately fairly new to golang and not a pro at this whole os process stuff.

Reply:
To clarify a bit: It looks like the child finishes initializing wayy before the worker starts working. I used prints to track what executes when to confirm when I first tried parent/child sync.